src/main/java/org/fogbowcloud/manager/core/models/orders/NetworkOrder.java

```diff
@@ -1,55 +1,55 @@
 package org.fogbowcloud.manager.core.models.orders;
 
-import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
+import java.util.UUID;
+
 import org.fogbowcloud.manager.core.models.token.Token;
 
 public class NetworkOrder extends Order {
 
-    private String gateway;
-    private String address;
-    private String allocation;
-
-    public NetworkOrder(OrderState orderState, Token localToken, Token federationToken, String requestingMember,
-                        String providingMember, OrderInstance orderInstace, long fulfilledTime, String gateway,
-                        String address, String allocation) {
-        super(orderState, localToken, federationToken, requestingMember, providingMember, orderInstace, fulfilledTime);
-        this.gateway = gateway;
-        this.address = address;
-        this.allocation = allocation;
-    }
-
-    public String getGateway() {
-        return gateway;
-    }
-
-    public void setGateway(String gateway) {
-        this.gateway = gateway;
-    }
+	private String gateway;
+	private String address;
+	private String allocation;
+
+	/**
+	 * Creating Order with predefined Id.
+	 */
+	public NetworkOrder(String id, Token localToken, Token federationToken, String requestingMember,
+			String providingMember, String gateway, String address, String allocation) {
+		super(id, localToken, federationToken, requestingMember, providingMember);
+		this.gateway = gateway;
+		this.address = address;
+		this.allocation = allocation;
+	}
 
-    public String getAddress() {
-        return address;
-    }
+	public NetworkOrder(Token localToken, Token federationToken, String requestingMember, String providingMember,
+			String gateway, String address, String allocation) {
+		this(UUID.randomUUID().toString(), localToken, federationToken, requestingMember, providingMember, gateway,
+				address, allocation);
+	}
 
-    public void setAddress(String address) {
-        this.address = address;
-    }
+	public String getGateway() {
+		return gateway;
+	}
 
-    public String getAllocation() {
-        return allocation;
-    }
+	public String getAddress() {
+		return address;
+	}
 
-    public void setAllocation(String allocation) {
-        this.allocation = allocation;
-    }
+	public String getAllocation() {
+		return allocation;
+	}
 
 	@Override
 	public OrderType getType() {
 		return OrderType.NETWORK;
 	}
 
+	/**
```

Comment:
Same comment ... I thought we planned to remove these methods. Please, remind me if we decided something else

Reply:
... or if my comments disagree with the design doc.

Reply:
Sorry, if you is talking about the getType method, this method is necessary for the Instance Provider. Will be necessary to identify the Order type. However, if you talking about the handleOpenOrder method, we already discussed it :smile:

Reply:
ok